agent-install/agent-install.sh

```diff
@@ -1670,29 +1357,30 @@ function cleanup_cluster_config_files() {
-
-    if [ "$USE_EDGE_CLUSTER_REGISTRY" == "true" ]; then
-			EDGE_CLUSTER_REGISTRY_PROJECT_NAME=$(echo $IMAGE_FULL_PATH_ON_EDGE_CLUSTER_REGISTRY|awk -F'/' '{print $2}')
-			EDGE_CLUSTER_AGENT_IMAGE_AND_TAG=$(echo $IMAGE_FULL_PATH_ON_EDGE_CLUSTER_REGISTRY|awk -F'/' '{print $3}')
-
-			if [[ "$INTERNAL_URL_FOR_EDGE_CLUSTER_REGISTRY" == "" ]]; then
-				# check if using local cluster or remote ocp
-				if $KUBECTL cluster-info | grep -q -E 'Kubernetes master.*//(127|172|10|192.168)\.'; then
-					# using small kube
-					sed -i -e "s#__ImagePath__#${IMAGE_FULL_PATH_ON_EDGE_CLUSTER_REGISTRY}#g" deployment.yml
-				else
-					# using ocp
-					IMAGE_FULL_PATH_ON_EDGE_CLUSTER_REGISTRY_INTERNAL_URL=$DEFAULT_INTERNAL_URL_FOR_EDGE_CLUSTER_REGISTRY/$EDGE_CLUSTER_REGISTRY_PROJECT_NAME/$EDGE_CLUSTER_AGENT_IMAGE_AND_TAG
-					sed -i -e "s#__ImagePath__#${IMAGE_FULL_PATH_ON_EDGE_CLUSTER_REGISTRY_INTERNAL_URL}#g" deployment.yml
-				fi
-			else
-				IMAGE_FULL_PATH_ON_EDGE_CLUSTER_REGISTRY_INTERNAL_URL=$INTERNAL_URL_FOR_EDGE_CLUSTER_REGISTRY/$EDGE_CLUSTER_REGISTRY_PROJECT_NAME/$EDGE_CLUSTER_AGENT_IMAGE_AND_TAG
-				sed -i -e "s#__ImagePath__#${IMAGE_FULL_PATH_ON_EDGE_CLUSTER_REGISTRY_INTERNAL_URL}#g" deployment.yml
-			fi		
+    sed -e "s#__AgentNameSpace__#${AGENT_NAMESPACE}#g" -e "s#__OrgId__#\"${HZN_ORG_ID}\"#g" deployment-template.yml >deployment.yml
+    chk $? 'creating deployment.yml'
+
+    if [[ "$USE_EDGE_CLUSTER_REGISTRY" == "true" ]]; then
+        EDGE_CLUSTER_REGISTRY_PROJECT_NAME=$(echo $IMAGE_FULL_PATH_ON_EDGE_CLUSTER_REGISTRY | awk -F'/' '{print $2}')
+        EDGE_CLUSTER_AGENT_IMAGE_AND_TAG=$(echo $IMAGE_FULL_PATH_ON_EDGE_CLUSTER_REGISTRY | awk -F'/' '{print $3}')
+
+        local image_full_path_on_edge_cluster_registry_internal_url
+        if [[ "$INTERNAL_URL_FOR_EDGE_CLUSTER_REGISTRY" == "" ]]; then
+            # check if using local cluster or remote ocp
+            if $KUBECTL cluster-info | grep -q -E 'Kubernetes master.*//(127|172|10|192.168)\.'; then
+                # using small kube
+                image_full_path_on_edge_cluster_registry_internal_url="$IMAGE_FULL_PATH_ON_EDGE_CLUSTER_REGISTRY"
+            else
+                # using ocp
+                image_full_path_on_edge_cluster_registry_internal_url=$DEFAULT_OCP_INTERNAL_URL_FOR_EDGE_CLUSTER_REGISTRY/$EDGE_CLUSTER_REGISTRY_PROJECT_NAME/$EDGE_CLUSTER_AGENT_IMAGE_AND_TAG
+            fi
+        else
+            image_full_path_on_edge_cluster_registry_internal_url=$INTERNAL_URL_FOR_EDGE_CLUSTER_REGISTRY/$EDGE_CLUSTER_REGISTRY_PROJECT_NAME/$EDGE_CLUSTER_AGENT_IMAGE_AND_TAG
+        fi
+        sed -i -e "s#__ImagePath__#${image_full_path_on_edge_cluster_registry_internal_url}#g" deployment.yml
     else
-		log_notify "Agent install on edge cluster requires to use an edge cluster registry"
-		exit 1
+        #lily: this makes it look like the only acceptable value for USE_EDGE_CLUSTER_REGISTRY is 'true'. If that is the case, why do we have this variable?
```

Comment:
So when started to implement agent install script for cluster install during last release, we had mentioned 2 ways for getting/storing agent image: 1) edge cluster image registry, 2) public registry. But later on we focused on edge cluster image registry and didn't talk about public registry anymore (also didn't mention use public registry in our doc), so the code is like only implemented the path that use the edge cluster image registry. 
So do we want to support public registry? I think we can get rid of `USE_EDGE_CLUSTER_REGISTRY` if we only want to support edge cluster that with image registry

Reply:
Thx, the background helps. I think we **do** want to support a public registry eventually. So we should keep the variable internally, but for now remove it from usage documentation. I can do that.